ui/app/js/controllers/release_new_controller.js

```diff
@@ -18,6 +18,28 @@ function($scope, $http, $modalInstance, CSRF, Releases, releases) {
   $scope.errors = {};
   $scope.saving = false;
 
+  $scope.fillName = function () {
+    var file = $scope.dataFile;
+
+    var reader = new FileReader();
+    reader.onloadend = function(evt) {
+      var blob = evt.target.result;
+      $scope.$apply( function() {
+         $scope.release.name = JSON.parse(blob).name;
+      });
+    };
+    if (typeof file !== 'undefined') {
+      // should work
+      reader.readAsText(file);
+    }
+
+  };
+
+  $scope.changeName = function () {
+    //wait for actual file to be loaded
+    setTimeout( $scope.fillName , 500);
```

Comment:
Turns out that Angular doesn't support onchange events to <input type="file">, per https://github.com/angular/angular.js/issues/1375, so this indirection is necessary.

Can you use 0 as the timeout instead of 500 though? The latter causes an unnecessary delay.